packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -741,12 +742,14 @@ class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
     setState(() {
       _candidateAvatars.remove(avatar);
     });
-    widget.onAccept?.call(avatar.data! as T);
-    widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
+    if (avatar.data != null)  {
+      widget.onAccept?.call(avatar.data! as T);
+      widget.onAcceptWithDetails?.call(DragTargetDetails<T>(data: avatar.data! as T, offset: avatar._lastOffset!));
+    }
```

Comment:
IMO this particular change has the potential to cause some bugs as well as some frustration during debugging code. Imagine you have a function A which you have defined inside these callbacks that are expected to be triggered when a Draggable is accepted. Now what this will do is entirely skip that trigger if your data is null. As a developer, it would be very difficult to pinpoint why this is exactly happening because it will not show any "info"/"warning" that these callbacks have not been triggered "because" data was null.

What are your views on this @Piinks ?

Reply:
@chinmoy12c For what it's worth, this change won't _"break"_ anything.
If this is merged as it is, `onAccept` and `onAcceptWithDetails` won't be called if `data` is `null`. And you might be right, it might not be the expected behavior.
However, as of today, it crashes when `data` is `null`, and therefore, `onAccept` and `onAcceptWithDetails` won't be called either.

So this _"will not make things worse"_. If we want to call `onAccept` and `onAcceptWithDetails` when `data` is `null`, I would have to change the method signatures and this would be a breaking change.

Reply:
> will not make things worse

I think embracing the yak shave is best in this case (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#lazy-programming)
In the linked PR (https://github.com/flutter/flutter/pull/133691) we're going to see if we can fix this inconsistency in the API

Reply:
Can you add documentation to the onAccept callback that mentions it will not be called if `data` null?
Then the `*WithDetails` variants can be called regardless of data being null once https://github.com/flutter/flutter/pull/133691 is finished.

Reply:
Sure, I added a sentence in [doc: Detail that onWillAccept and onWillAcceptWithDetails won't be called if data is null](https://github.com/flutter/flutter/pull/133136/commits/17581cc2b43c0bffa4c2cf1dcd112e2dee10605d)